src/coreclr/gc/gc.cpp

```diff
@@ -21752,6 +21752,7 @@ void gc_heap::gc1()
     {
 #ifdef USE_REGIONS
         initGCShadow();
+        rearrange_uoh_segments();
```

Comment:
Ooops, this needs to be done for segment as well, need to move slightly earlier before the `ifdef`.